src/components/TracingVisualization/TracingVisualizationRow.tsx

```diff
@@ -0,0 +1,114 @@
+import { debounce } from "ts-debounce";
+import React, { useLayoutEffect, useMemo, useRef, useState } from "react";
+import { nsToMs } from "../../helpers/nsToMs";
+
+interface ITracingVizualizationRowProps {
+  name?: string;
+  type?: string;
+  color?: 'green' | 'purple' | 'indigo';
+  total: number;
+  offset?: number;
+  duration: number;
+}
+
+export const TracingVisualizationRow = (props: ITracingVizualizationRowProps) => {
+  const { name, total, offset, duration, type, color } = props;
+
+  const backgroundColorCss = getBackgroundColors(color || type);
+
+  const container = useRef<HTMLDivElement>(null)
```

Comment:
This code looks like it could be really useful for generic cases when needing to resize. Could we extract to a reusable hook within the `src/hooks` folder?

I was thinking something like:

```
const ref = useResize({ initialWidth: 250 })
```

Reply:
I originally did pull it out but it wasn't working as expected and I couldn't understand why. And I now forget what the actual error was.

I'll try again.

Reply:
I created useBoudingRect.